### ISSUE TITLE
Timeout prevents application exit

### ISSUE THREAD

**Issue by onitake** [sparrc/go-ping#11]
When sending an unsuccessful ping request, it is no longer possible to end an application in a proper fashion, i.e. Ctrl-C does not work. I suspect the timeout code does not end all goroutines properly.

Example code:
```
	pinger, err := ping.NewPinger(addr)
	if err != nil {
		log.Fatal(err)
	}
	pinger.Timeout = time.Seconds * 10
	pinger.Count = 3
	pinger.SetPrivileged(true)
	pinger.Run()
	stats := pinger.Statistics()
	log.Println(stats)
```

**Comment by devinodaniel** [comment]
Also experiencing this. @onitake did you ever trace or confirm that `pinger.Timeout` is the issue?

**Comment by onitake** [comment]
I didn't; for my use case, I opted for a simpler approach with an exec of `ping` instead.
This has the advantage of not requiring superuser privileges. The disadvantage is that you need to depend on a minimal shell environment.
But it should be fairly easy to trace with pprof.

**Comment by jmunsin** [comment]
I had a similar problem, the following change seems to fix it:

diff --git a/ping.go b/ping.go
index c8cae5d..2cd213a 100644
--- a/ping.go
+++ b/ping.go
@@ -289,6 +289,7 @@ func (p *Pinger) run() {
        c := make(chan os.Signal, 1)
        signal.Notify(c, os.Interrupt)
        signal.Notify(c, syscall.SIGTERM)
+       defer signal.Stop(c)

        for {
                select {

**Comment by robisonjoel** [comment]
I tried jmunsin's fix and con confirm it has the desired results

**Comment by MichaelHipp** [comment]
I can confirm this problem also. Nearly impossible to make a program exit once it ping has been Run.

**Comment by onitake** [comment]
@jmunsin Can you turn this into a PR and submit it?
This is something that should finally be fixed...

**Comment by sparrc** [comment]
Thank you for the report. The library was erronously previously capturing the ctrl-C signal. This was fixed by #37
